### ISSUE TITLE
Pneumatic Transport Tubes

### ISSUE THREAD

**Issue by SwissxCore** [TeamPneumatic/pnc-repressurized#1198]
### Describe the feature

(Apologies if this has been suggested, I've tried searching for closed topics pertaining to this, but couldn't see any)

As a player becomes situated in their world and begins to expand their operations across many chunks, they tend to find themselves running back and forth more often, whether they have forgotten something they needed, or their machines and contraptions aren't yet integrated together, or perhaps they have planned their base poorly (which I'm often guilty of, for sure), it can become tiresome fast; the solution? Transport.

Now, Minecraft has multiple methods of transportation, but they often have their own restrictions; A horse can travel fast and jump high with good genetics, but they tend to wander, and require larger passageways. A minecart can travel relatively fast, but are very vertically challenged; if you base has multiple levels, you better get to work creating a staircase for your tracks. A boat on ice can travel extremely fast but can't go upwards; a nether portal can shorten the distance, but still takes time, and it's dangerous in there.

No matter the method of transportation, none would compare to the absolute majesty of a dynamic pneumatic transport tube; it will allow the player to travel fast, no matter the direction; players on multiplayer servers can also connect their bases using this method; and assuming the tubes a glass, it would provide a pretty nice view as well.

So, would this make anything else in the mod obsolete? The truth is, it depends; one could simply use this in place of an elevator, but there's a catch, the pneumatic transport tubes can only connect to two points, so you can have an elevator, but only to one floor up; not to mention the air cost would be astronomical compared to a traditional pneumatic elevator. The transport tubes aren't able to transport items, nor at they able to be used by drones for logistics, so air cannons would still be useful, and drones won't become an unstoppable force as you would still require multiple hubs for them to function within the loaded chunks.

All in all, the transport tubes would allow for a player to get from point A to B in record time to cut down on the monotony of travel.

In order to use the pneumatic transport tubes, the player will require two transport docks, these docks are responsible for sending and receiving the player; the tubes themselves will be connected on the top or bottom of the dock, affording the player two connection points to support their base layout, but only one point can be used at a time. The back of the dock will connect to a vacuum pump (or to whatever setup you have, as long as you can achieve a vacuum), this is because the transport tubes will rely entirely on a vacuum, so any pressure above 1 bar simply isn't necessary.

Unfortunately, the player wouldn't just simply walk into the dock, but would instead right click on it to enter; at this stage, the door will seal the player inside, and once the vacuum has been reached in the system, the player would press "space" to begin travelling; the hatch would snap open sucking the player through the system until they reach the other dock; if the system hasn't reached the correct negative pressure yet, pressing "space" will do nothing, but the player can press "left shift" at any point to exit the dock, even once it's ready, pressing "left shift" instead of "space" will still dismount the player.

Lastly, once the system has been used, it will be fully pressurised again, this is because the pressure differential is what allows the player to move through the transport tubes; as a result, the system will need to be vacuumed again before it can be used. The player may choose to build two of these systems side by side, but redstone can be used to flip which side begins depressurising the system ready for transport.

Thank you.

### Reasons why it should be considered

This mod offers many fun features for the player to play with, but besides the pneumatic suit (which I assume, without any speed upgrades, would be slow than these tubes), or the OSHA non-compliant vortex cannon, the mod fails to offer any external means of player transportation.

The current vanilla Minecraft transportation options are either unable to perform the same feats a pneumatic transport tube would achieve, require some complex technical redstone knowhow, or would need the player to exploit the game in some way.

I love PneumaticCraft, but I fear it has been starving for a major content update for a while now, and after recalling these very same transport systems used in Futurama, I felt, with the theme of the mod, it would be a great addition to the player's creative capabilities.

### Additional details

Using dye, the docks can be colour coded, although two connected docks don't require the same colour. Additionally, docks don't have to be coloured at all.

Having the player right click to enter the dock would probably make this much easier on your end to code, but would probably mean the dock, despite being two blocks tall (maybe more), would be placed in one go as I'm not sure it would work well as a multiblock structure.

Using the pneumatic wrench, the player can alternate the tubes between having solid walls, and transparent glass; this is to avoid multiple recipes for the transport tubes.

The a pneumatic transport system only connects from point A to point B, there can be no junctions as it would interfere with the vacuum, thus using the pneumatic wrench to rotate tubes would be unnecessary, allowing the function to be used for switching between glass and back instead.

I was thinking the tubes would be able to pass through nether portals and continue onwards just as Create allows trains to pass through them, but I feel there would be no way to achieve this without significantly interrupting the immersion.

**Comment by desht** [comment]
It's a nice idea but unlikely to be implemented any time soon, simply due to the size of the task, and the amount of time I actually have.

You did miss one transport option that PNC has, though: drones.  Drones can pick players up and fly them to nearby locations (or teleport them to distant locations, if the server config allows it).

**Comment by SwissxCore** [comment]
I had no idea the drones could transport players around, that's pretty cool actually. I've always found drones to be difficult to get into personally, even with the puzzle-based programming, I tend to lose track of what the drone is supposed to be doing and when, and given how hard it is to get plastic early on, making mistakes with the programming can be costly, so I just usually avoid them. Conversely I don't mind using the specialised drones since they're already pre-programmed to perform their tasks without my input; I find those are fine.

I appreciate the response however; it certainly would be a major task and I wouldn't fault you for refusing in the end, the mod is already excellent as it is and wouldn't necessarily lose anything by not having these transport tubes.

**Comment by desht** [comment]
Programming mistakes shouldn't be costly, since it's always possible to take puzzle pieces out of a programmed drone (just write an empty "program" onto a programmed drone, and you'll get all your pieces back).  Also, creative is a thing when you're learning :)

As for the transport tubes, I do like the idea so I'm not saying no.  Just not in a position to implement anything major like that at the moment...

**Comment by SwissxCore** [comment]
You are correct, I could play around with drones in creative; I can't really remember the last time I did anything in creative, but it would certainly help. I suppose, as I said, I've never really gotten into drones, so I shouldn't be talking about them as if I know anything, at that point it's just misinformation.

And I absolutely understand, all in all, I'm glad you like the suggestion.

**Comment by johalun** [comment]
Futurama transport tubes would be pretty cool but a huge task to implement for sure :D
